gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -293,4 +293,187 @@ fluid.registerNamespace("gpii.transformer");
+     *     }
+     *  ]
+     *
+     * @param {Array} arr An array with the elements that will be transformed into the tagged array.
+     * @param {Object} transformSpec The required spec for the transform, it should have the form:
+     *  {
+     *     "tag": "tagValue",
+     *     "key": "keyValue",
+     *     "elems": [
+     *     ],
+     *     "hasElemVal": {
+     *         "isTrue": "isTrueValue",
+     *         "isFalse": "isFalseValue"
+     *     }
+     *   }
+     * @return {Array} A tagged array result of processing the input array with the provided 'transformSpec'.
+     */
+    gpii.transformer.jsonArrayToTaggedArray = function (arr, transformSpec) {
+        var output = [];
+
+        // Check the transformSpec
+        gpii.transformer.jsonArrayToTaggedArray.checkSpec(transformSpec);
+
+        fluid.each(transformSpec.elems, function (elem) {
+            var arrayElem = {};
+
+            var hasElem = arr.indexOf(elem) !== -1;
+
+            arrayElem[transformSpec.key] = elem;
+            arrayElem[transformSpec.tag] = transformSpec.hasElemVal[hasElem ? "isTrue" : "isFalse"];
+
+            output.push(arrayElem);
+        });
+
+        return output;
+    };
+
+    /**
+     * Checks that the provied specification have the proper form, if the provided spec doesn't
+     * have the correct form, 'fluid.fail()' will be called.
+     *
+     * @param {Object} transformSpec The transform spec to be validated.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.checkSpec = function (transformSpec) {
+        if (!transformSpec.tag) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'tag' block set");
+        }
+        if (!transformSpec.key) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'key' block set");
+        }
+        if (!transformSpec.elems || !fluid.isArrayable(transformSpec.elems)) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'elems' block array");
+        }
+        if (transformSpec.hasElemVal) {
+            if (!transformSpec.hasElemVal.isTrue) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isTrue' value set");
+            }
+            if (!transformSpec.hasElemVal.isFalse) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isFalse' value set");
+            }
+        } else {
+            fluid.fail("jsonArrayToTaggedArray requires a 'hasElemVal' set");
+        }
+    };
+
+    /**
+     * Defined the inverse transform for 'gpii.transformer.jsonArrayToTaggedArray'.
+     *
+     * @param {Object} transformSpec The transform spec which inverse is going to be defined.
+     * @return {Object} The modified transform spec.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.invert = function (transformSpec) {
+        transformSpec.type = "gpii.transformer.taggedArrayToJSON";
+        return transformSpec;
+    };
+
+    fluid.defaults("fluid.transforms.taggedArrayToJSON", {
```

Comment:
Wrong namespace used here - you were saved by the fact that this is the default grade used if none is found

Reply:
Nice catch, fixed!